ykrt/src/profile/mod.rs

```diff
@@ -0,0 +1,34 @@
+//! Profiler support for JITted code.
+//!
+//! This module specifies the platform abstraction for trace profiling support.
+
+use crate::compile::CompiledTrace;
+use std::{error::Error, sync::Arc};
+
+#[cfg(target_os = "linux")]
+mod linux_perf;
+
+pub(crate) trait PlatformTraceProfiler: Send + Sync {
+    /// Register newly JITted trace with the platform's profiler.
+    fn register_ctr(&self, ctr: &Arc<dyn CompiledTrace>) -> Result<(), Box<dyn Error>>;
+}
+
+/// Profiler support that does nothing.
+///
+/// Used when the current platform has no profiler support implemented, or when the user has not
+/// turned on profiling for this run.
+struct DummyProfiler {}
```

Comment:
I think `NullProfiler` might be a better name? [Bikeshedding I know, but "dummy" has been used elsewhere in yk to mean "only useful in tests".]